bazaar/interface.rst

```diff
@@ -1,12 +1,207 @@
+   | - Drag and Drop
+   | - Window to window
+   | NOTE: These material selection methods cannot be used from mobile devices (Windows Phone, iOS, Android). 
+3. User selects material from integrated embedded Edustore JS module, which can be included to any LMS. This method uses drag and drop.
+
+
+API workflow for selecting material from Edustore directly:
+
+1. LMS makes a JSON POST request to the Edustore with params from table 1, using header-based auth.
+2. Edustore replies with unique URL and LMS forwards user to the unique url.  
+3. User selects learning material from the Edustore and Edustore forwards user back to the add_resource_callback_url, with resource details(ie. name, description).
+
+Params table 1:
+
+==========================  ================================================ ======= =========
+Param                       Description                                      Type    Required
+==========================  ================================================ ======= =========
+first_name                  First name of user.                              alpha   required
+last_name                   Last name of user.                               alpha   required
+email                       Email of user.                                   email   optional
+user_id                     User id from                                     alpha   required
+context_id                  Course or page identifier. Must be unique.       alpha   required
+context_title               Course or page title                             alpha   required
+role                        Role (enum student, teacher, admin)              alpha   required
+school                      Name of the school                               alpha   required
+school_id                   School's identifier in the national level.       alpha   required
+city                        City's name                                      alpha   required
+city_id                     City's unique identifier in the national level   alpha   required
+oid                         User oid                                         alpha   optional
+add_resource_callback_url   LMS callback URL, for adding material to the LMS url     optional
+cancel_url                  Forwarding user back to  LMS                     url     optional
+==========================  ================================================ ======= =========
+
+
+1.1 Material drag and drop
+--------------------------
+
+LMS should implement drop functionality directly and parse links with "edustore.fi/<base64 encoded json>.
+Base64 encoded JSON has basic information about the resource: name, description and uid.
+
+
+2 Viewing material 
+===================
+
+Use case:
+Person wants to view selected material/resource. 
+
+API workflow:
+
+1. LMS makes a JSON POST request to the Edustore with params from table 2, using header-based auth.
```

Comment:
It says here that only header-based auth is supported, but at the top of the document two auth methods are introduced.